core/domain/exp_domain.py

```diff
@@ -779,12 +860,24 @@ def validate(self, exp_param_specs_dict):
         for fallback in self.fallbacks:
             fallback.validate()
 
+        if not isinstance(self.hints, list):
+            raise utils.ValidationError(
+                'Expected hints to be a list, received %s'
+                % self.hints)
+        for hint in self.hints:
+            hint.validate()
+
+        if not isinstance(self.solution, dict):
```

Comment:
Are you planning to validate the Solution object?

Also, shouldn't self.solution be a Solution object, not a dict?